IntegrationAdapters/Controllers/MedicineController.cs

```diff
@@ -1,35 +1,78 @@
 ﻿using IntegrationAdapters.Models;
+using IntegrationAdapters.Models.DTO;
 using IntegrationAdapters.Services;
 using Microsoft.AspNetCore.Mvc;
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
-using System.Threading.Tasks;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
 
 namespace IntegrationAdapters.Controllers
 {
     [Route("[controller]")]
     [ApiController]
-    public class MedicineController
+    public class MedicineController : ControllerBase
     {
         private readonly MedicineService medicineService;
 
         public MedicineController()
         {
             this.medicineService = new MedicineService();
         }
-        [HttpGet("getMedicineSpecification/{MedicineName}")]
-        public String GetMedicineSpeification(String MedicineName)
+        [HttpGet("getMedicineSpecification/{medicineName}")]
+        public IActionResult GetMedicineSpeification(String medicineName)
         {
-            Medicine medicine = medicineService.GetMedicineByName(MedicineName);
-            if (medicineService.GetMedicineByName(MedicineName) != null)
+            Medicine medicine = medicineService.GetMedicineByName(medicineName);
+            if (medicineService.GetMedicineByName(medicineName) != null)
```

Comment:
The same method is called two times here, this could be just one method call service.DoesMedicineExist(medicineName) so the service does the null check